tensorflow/cc/gradients/functional_grad_test.cc

```diff
@@ -0,0 +1,84 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/cc/framework/ops.h"
+#include "tensorflow/cc/ops/array_ops.h"
+#include "tensorflow/cc/ops/functional_ops.h"
+#include "tensorflow/cc/ops/standard_ops.h"
+#include "tensorflow/core/framework/function_testlib.h"
+#include "tensorflow/cc/framework/grad_op_registry.h"
+#include "tensorflow/cc/framework/gradient_checker.h"
+#include "tensorflow/cc/framework/testutil.h"
+#include "tensorflow/cc/gradients/grad_testutil.h"
+#include "tensorflow/core/framework/tensor_testutil.h"
+#include "tensorflow/core/lib/core/status_test_util.h"
+
+namespace tensorflow {
+namespace {
+
+using namespace ops;  // NOLINT(build/namespaces)
```

Comment:
Clang tidy doesn't like this ([neither does the style guide](https://google.github.io/styleguide/cppguide.html#Namespaces)). Should this test just be in the ops namespace too?

Reply:
Seems fine, I just copied it from the other tests.

Should I do this (and add the build dependency) for #46115 too?

Reply:
Ugh. Yeah, looks like that's why that one hasn't submitted itself either yet. The suggestions from the automated tool for that one:

  Add //tensorflow/cc:gradients
  Add //tensorflow/cc:ops
  Add //tensorflow/cc:client_session

(to gradients_resource_variable_grad_test)

Very annoying when the internal CI is stricter than the checks we have here.